test/src/test/java/hudson/model/queue/WideExecutionTest.java

```diff
@@ -87,6 +95,33 @@ public String getDisplayName() {
         }
     }
 
+    @Issue("JENKINS-30084")
+    @Test
```

Comment:
`QueueTest` might be a more natural place to put this, since it already has `flyweightTasks` and some related tests. This test suite seems to be dedicated to `SubTaskContributor`, which is unrelated.